tests/unit/linalg/test_ode2.cpp

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) 2010, Lawrence Livermore National Security, LLC. Produced at
+// the Lawrence Livermore National Laboratory. LLNL-CODE-443211. All Rights
+// reserved. See file COPYRIGHT for details.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability see http://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the GNU Lesser General Public License (as published by the Free
+// Software Foundation) version 2.1 dated February 1999.
+//
+
+#include "mfem.hpp"
+#include "catch.hpp"
+#include <cmath>
+
+using namespace mfem;
+
+TEST_CASE("Second order ODE methods",
+          "[ODE2]")
+{
+   double tol = 0.25;
+
+   /** Class for simple linear second order ODE.
+    *
+    *    du2/dt^2 + b du/dt  + a u = 0
+    *
+    */
+   class ODE2 : public TimeDependentOperator
+   {
+   protected:
+      double a,b;
+
+   public:
+      ODE2(double a_, double b_) : TimeDependentOperator(1, 0.0),  a(a_), b(b_) {};
+
+      virtual void ExplicitSolve(const Vector &u, const Vector &dudt,
+                                 Vector &d2udt2)  const
+      {
+         d2udt2[0] = -a*u[0] - b*dudt[0];
+      }
+
+      virtual void ImplicitSolve(const double fac0, const double fac1,
+                                 const Vector &u, const Vector &dudt, Vector &d2udt2)
+      {
+         double T = 1.0 + a*fac0 + fac1*b;
+         d2udt2[0] = (-a*u[0] - b*dudt[0])/T;
+      }
+
+      virtual ~ODE2() {};
+   };
+
+   /** Class for checking order of convergence of first order ODE.
+    */
+   class CheckODE2
+   {
+   protected:
+      int ti_steps,levels;
+      Vector u0;
+      Vector dudt0;
+      double t_final,dt;
+      ODE2 *oper;
+   public:
+      CheckODE2()
+      {
+         oper = new ODE2(1.0, 0.0);
+         ti_steps = 40;
+         levels   = 5;
```

Comment:
I recommend changing this to 7 to show more conclusive (at least to me) convergence rate behavior: with `levels=5` the convergence rate for some methods only reached the expected value on the last run. 

Of course, an alternative to this would be to keep `levels=5` and change `ti_steps=160`

Reply:
Due to rate vs du fix from above I can even reduce to ti_steps=20 I believe.

Reply:
Your rate question from above pointed me to a start-up issue, similar as for the genalpha for the first order problem. This fixed the not quite nice convergence.

Reply:
Great! The convergence test numbers look much better now.